src/tendencies/hyperdiffusion.jl

```diff
@@ -104,11 +110,22 @@ function hyperdiffusion_tendency!(Yₜ, Y, p, t)
         end
     end
 
-    @. Yₜ.c.uₕ -= κ₄ * divergence_damping_factor * C12(wgradₕ(divₕ(ᶜ∇²uₕ)))
+    @. ᶜ∇⁴ᶜu = divergence_damping_factor * C123(wgradₕ(divₕ(ᶜ∇²ᶜu)))
+    if point_type <: Geometry.Abstract3DPoint
+        ᶜ∇²ᶜu₁₂ = @. Geometry.C12(ᶜ∇²ᶜu)
+        ᶜ∇²ᶜu₃ = @. Geometry.C3(ᶜ∇²ᶜu)
+        # do the curl
+        ᶜχ₁₂₃ = @. Geometry.C123(scurl(ᶜ∇²ᶜu₁₂)) + Geometry.C123(scurl(ᶜ∇²ᶜu₃))
+        ᶜχ₁₂ = @. Geometry.C12(ᶜχ₁₂₃)
+        ᶜχ₃ = @. Geometry.C3(ᶜχ₁₂₃)
+        # do the second curl
+        @. ᶜ∇⁴ᶜu -= Geometry.C123(wcurl(ᶜχ₁₂)) + Geometry.C123(wcurl(ᶜχ₃))
+    end
+
+    @. Yₜ.c.uₕ -= κ₄ * C12(ᶜ∇⁴ᶜu)
     # Without the C12(), the right-hand side would be a C1 or C2 in 2D space.
     if point_type <: Geometry.Abstract3DPoint
-        # TODO: Are these vector conversions correct when there is topography?
-        @. Yₜ.c.uₕ += κ₄ * C12(wcurlₕ(C3(curlₕ(ᶜ∇²uₕ))))
+        @. Yₜ.c.uₕ -= κ₄ * IC2F(C3(ᶜ∇⁴ᶜu))
```

Comment:
Return type here is inconsistent with lhs I think; based on above lines `ᶜ∇⁴ᶜu` is `C123`, lhs is `C12`, rhs is `C3`